core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java

```diff
@@ -118,6 +112,22 @@ public void write(String s, byte[] bytes) {
                 greaterEqual24_2.invokeStaticMethod(REGISTER_RUNTIME_SYSTEM_PROPERTIES,
                         greaterEqual24_2.load("user.language"),
                         greaterEqual24_2.load("en"));
+            }
+        }
+        // The deprecated option takes precedence for users who are already using it.
+        if (nativeConfig.userCountry().isPresent()) {
```

Comment:
Why exactly did you change that? Because for me, either you use the deprecated `nativeConfig.userLanguage` and then you also use `nativeConfig.userCountry` or you're using the new stuff.

I think the previous way of doing things was perfectly fine.

Reply:
In the previous version we would only respect userCountry if userLanguage was also used. With the new version the behavior is more consistent with the existing one https://github.com/quarkusio/quarkus/blob/2efe0aa43c0aeb55818d5077f2c52e61e531c0b3/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java#L750-L757 and https://github.com/quarkusio/quarkus/blob/c025a1b29c02b6a4b8b773d6041c9e50119147a0/core/deployment/src/main/java/io/quarkus/deployment/steps/LocaleProcessor.java#L89-L124

Reply:
In other words this change is indeed not necessary to address your comment, but I feel like it's an improvement. Happy to revert if you prefer it the previous way.